### ISSUE TITLE
How to set the skeleton of an application with FastAPI + Typer

### ISSUE THREAD

**Issue by jorgerpo** [tiangolo/fastapi#1115]
Hello
Has anybody used Typer + FastAPI yet to get something similar to **goxygen**?
https://github.com/Shpota/goxygen
"
Generate a Web project with Go, Angular/React/Vue, and MongoDB in seconds.

Goxygen aims at saving your time while setting up a new project. It creates a skeleton of an application with all configuration done for you. You can start implementing your business logic straight away. Goxygen generates back end Go code, connects it with front end components, provides a Dockerfile for the application and creates docker-compose files for convenient run in development and production environments."

I'd like to set up a Crud in fastapi with SqlAlchemy and  with Vuetify for frontend.

example of skelleton with goxygen:

`Structure of a generated project (React-based app)

my-app
├── server                   # Go project files
│   ├── db                   # MongoDB communications
│   ├── model                # domain objects
│   ├── web                  # REST APIs, web server
│   ├── server.go            # the starting point of the server
│   └── go.mod               # server dependencies
├── webapp                    
│   ├── public               # icons, static files, and index.html
│   ├── src                       
│   │   ├── App.js           # the main React component
│   │   ├── App.css          # App component-specific styles
│   │   ├── index.js         # the entry point of the application          
│   │   └── index.css        # global styles
│   ├── package.json         # front end dependencies
│   ├── .env.development     # holds API endpoint for dev environment
│   └── .env.production      # API endpoint for prod environment
├── Dockerfile               # builds back end and front end together
├── docker-compose.yml       # prod environment deployment descriptor
├── docker-compose-dev.yml   # runs local MongoDB for development needs
├── init-db.js               # creates a MongoDB collection with test data
├── .dockerignore            # specifies files ignored in Docker builds
├── .gitignore
└── README.md                # guide on how to use the generated repo
`

**Comment by Dustyposa** [comment]
how about this one?
[https://fastapi.tiangolo.com/project-generation/](https://fastapi.tiangolo.com/project-generation/)
or
this one
[https://github.com/nsidnev/fastapi-realworld-example-app](https://github.com/nsidnev/fastapi-realworld-example-app)

**Comment by jorgerpo** [comment]
Yes. Right.
But what if you want to personalize even more.
Like you don't want to use vue but react, etc.

I see fastAPI + Typet can accompish something similar to Vue + Nuxt + Vue CLI

I wonder if anybody has some pointers.

**Comment by tiangolo** [comment]
Yeah, that could be interesting, but it would be a whole new (and quite a big) project.

I don't think anyone has done something similar, as that would probably cover/overlap some of the things that Cookiecutter is already doing, and they are currently improving it a lot and adding features, etc.

I think the closest we'll have to something like that would actually be the improvements in Cookiecutter and updates in the current generators to use those new things.
